pkg/block/fetcher.go

```diff
@@ -574,4 +602,49 @@ func (f *ConsistencyDelayMetaFilter) Filter(metas map[ulid.ULID]*metadata.Meta,
 			delete(metas, id)
 		}
 	}
+
+	return nil
+}
+
+// IgnoreDeletionMarkFilter is a MetaFetcher that filters blocks that are marked for deletion; delay duration before current time.
+// The delay duration is to make sure that the replacement block can be fetched before we filter out the old block.
+type IgnoreDeletionMarkFilter struct {
+	logger          log.Logger
+	delay           float64
```

Comment:
time.Duration would be nicer, so we know the units.

Reply:
I have made this change. Thanks for pointing out.